core-libs/setup/ssr/optimized-engine/optimized-ssr-engine.ts

```diff
@@ -525,20 +509,26 @@ export class OptimizedSsrEngine {
    * @private
    */
   private updateWithRequestContext(options: any) {
-    const [requestContext, error] = this.getRequestContext(options.req);
+    const requestContext = this.getRequestContext();
     options.req.res.locals = {
       cx: { request: requestContext },
     };
-    if (error) {
-      this.logger.error(error.message, { request: options.req });
-    }
   }
 
   //CXSPA-3680 - remove this method in 7.0
   private initLogger(ssrOptions: SsrOptimizationOptions | undefined) {
-    if (ssrOptions?.logger === true) {
-      return new DefaultExpressServerLogger();
+    const logTransformers = getLogTransformers(
+      this.ssrOptions?.logTransformers,
+      defaultTransformersConfig
+    );
+    if (ssrOptions?.logger) {
+      const logger =
+        ssrOptions?.logger === true
+          ? new DefaultExpressServerLogger()
+          : ssrOptions.logger;
+      logger.setTransformers(logTransformers);
+      return logger;
     }
-    return ssrOptions?.logger || new LegacyExpressServerLogger();
+    return new LegacyExpressServerLogger();
   }
 }
```

Comment:
> `//CXSPA-3680 - remove this method in 7.0`

Now this method seems to be needed. So we should move and change this code comment, to describe more precisely what should be done as part of CXSPA-3680, instead of removing the whole method.